### DISCUSSION TITLE
What is RecordType?

### DISCUSSION THREAD

**Discussion by chenbj5515** [ant-design/ant-design#30287] [Q&A]
I can find a lot of RecordType in antd's .d.ts files, but I can't figure out what it is. It's a type, a interface or sth else?where can i find the place it be defined?

**Comment by lamhungypl** [comment]
It's generic type.
https://www.typescriptlang.org/docs/handbook/2/generics.html

**Reply by steveharrisondcs** [reply]
Can't see `RecordType` mentioned anywhere in that link?

**Comment by DBSDs** [comment]
It's generic type.
You can use 'T' or 'U' and anything else to replace 'RecordType'.
In javascript it's like a variable: let a =..., but in typescript let is not required.
for example, you can define a  same generic type with diffrent variable name.
1. type T1\<RecordType extends any\> = RecordType.
2. type T2\<T extends any\> = T .

they are equal.
